### ISSUE TITLE
[BUG] Node Entirely Broken After `CloudInit` Resource Applied

### ISSUE THREAD

**Issue by irishgordo** [harvester/harvester#5119]
**Describe the bug**
Applying a CloudInit CRD yaml that contains multiple negative based entries as well as some positive, encouraged ones, breaks the node entirely.
Things noticed:
- user is locked out of the node via trying to just utilize the set password
- the hostname on the node changes

**To Reproduce**
1. Provision a master-head ISO based install (I used 2/5)
1. Create a CRD YAML (see CRD YAML section)
1. Apply the CRD YAML
1. Reboot the node
1. The node remains in a broken state, seems to not recover, things noticed ( hostname changed & user password fails to work )

**Expected behavior**
The WebHook protections in place preventing the node with mallicious CloudInit applied from getting into a strange place

**Support bundle**
Node is inaccessible 

**Environment**
 - Harvester ISO version: master-head (2/5) && v1.3.0-rc1 
 - Underlying Infrastructure: HP DL160 Gen9 && QEMU/KVM

**Additional context**
CRD YAML Example:
```
apiVersion: node.harvesterhci.io/v1beta1
kind: CloudInit
metadata:
  name: ssh-access
spec:
  matchSelector: {}
  filename: 99_ssh.yaml
  contents: |
    stages:
      network:
        - authorized_keys:
            rancher:
              - ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIBzZT+yXkr28BJzki4WdisefgyR1hKMXWlJCd9KfajEm michael.russell@suse.com
---
apiVersion: node.harvesterhci.io/v1beta1
kind: CloudInit
metadata:
  name: grubcustom
spec:
  matchSelector: {}
  filename: grubcustom
  contents: |
    hello there
---
apiVersion: node.harvesterhci.io/v1beta1
kind: CloudInit
metadata:
  name: 99settings
spec:
  matchSelector: {}
  filename: 99_settings.yaml
  contents: |
    hello there
---
apiVersion: node.harvesterhci.io/v1beta1
kind: CloudInit
metadata:
  name: 90custom
spec:
  matchSelector: {}
  filename: 90_custom.yaml
  contents: |
    hello there
---
apiVersion: node.harvesterhci.io/v1beta1
kind: CloudInit
metadata:
  name: elemental.config
spec:
  matchSelector: {}
  filename: elemental.config
  contents: |
    hello there
---
apiVersion: node.harvesterhci.io/v1beta1
kind: CloudInit
metadata:
  name: grubenv
spec:
  matchSelector: {}
  filename: grubenv
  contents: |
    hello there
---
apiVersion: node.harvesterhci.io/v1beta1
kind: CloudInit
metadata:
  name: harvesterconfig
spec:
  matchSelector: {}
  filename: harvester.config
  contents: |
    hello there
---
apiVersion: node.harvesterhci.io/v1beta1
kind: CloudInit
metadata:
  name: install
spec:
  matchSelector: {}
  filename: install
  contents: |
    hello there
---
apiVersion: node.harvesterhci.io/v1beta1
kind: CloudInit
metadata:
  name: consolelogoverwrite
spec:
  matchSelector: {}
  filename: install/console.log
  contents: |
    hello there
```

**Comment by connorkuehl** [comment]
Fix should be available in 1.3.0-rc3, moving to Ready for Testing.

**Comment by irishgordo** [comment]
@connorkuehl  this looks great in rc3!
Thanks for the fix!
:smile: :+1: 
```
╭─mike at suse-workstation-team-harvester in ~/Projects/cloud-init-test-4710
╰─○ kubectl create -f test.yaml 
cloudinit.node.harvesterhci.io/ssh-access created
Error from server (InternalError): error when creating "test.yaml": admission webhook "validator.harvester-system.harvester-node-manager-webhook" denied the request: Internal error occurred: could not parse document as yaml
Error from server (InternalError): error when creating "test.yaml": admission webhook "validator.harvester-system.harvester-node-manager-webhook" denied the request: Internal error occurred: filename conflicts with a critical system-owned file
Error from server (InternalError): error when creating "test.yaml": admission webhook "validator.harvester-system.harvester-node-manager-webhook" denied the request: Internal error occurred: filename conflicts with a critical system-owned file
Error from server (InternalError): error when creating "test.yaml": admission webhook "validator.harvester-system.harvester-node-manager-webhook" denied the request: Internal error occurred: could not parse document as yaml
Error from server (InternalError): error when creating "test.yaml": admission webhook "validator.harvester-system.harvester-node-manager-webhook" denied the request: Internal error occurred: could not parse document as yaml
Error from server (InternalError): error when creating "test.yaml": admission webhook "validator.harvester-system.harvester-node-manager-webhook" denied the request: Internal error occurred: could not parse document as yaml
Error from server (InternalError): error when creating "test.yaml": admission webhook "validator.harvester-system.harvester-node-manager-webhook" denied the request: Internal error occurred: could not parse document as yaml
Error from server (InternalError): error when creating "test.yaml": admission webhook "validator.harvester-system.harvester-node-manager-webhook" denied the request: Internal error occurred: could not parse document as yaml
╭─mike at suse-workstation-team-harvester in ~/Projects/cloud-init-test-4710
╰─○ ssh rancher@192.168.122.28
rancher@testv130rc3:~> exit
logout
Connection to 192.168.122.28 closed.

```
